Source/CrossingDetector/CrossingDetector.cpp

```diff
@@ -415,7 +392,7 @@ bool CrossingDetector::shouldTrigger(const float* rpCurr, int nSamples, int t0,
     int pastSamplesNeeded = pastSpan ? static_cast<int>(ceil(pastSpan * pastStrict)) : 0;
     int futureSamplesNeeded = futureSpan ? static_cast<int>(ceil(futureSpan * futureStrict)) : 0;
     // if enough values cross threshold
-    if(posOn)
+    if(currPosOn)
```

Comment:
This was a bug - probably caused the weird results when both "rising" and "falling" were selected.